### ISSUE TITLE
Possible bugs in sample_weight in model.fit() for RNN

### ISSUE THREAD

**Issue by blackyang** [fchollet/keras#657]
Hi there,

From Keras's documentation it says that:

>For time-distributed data, there is one weight per sample per timestep, i.e. if your output data is shaped (nb_samples, timesteps, output_dim), your mask should be of shape (nb_samples, timesteps). This allows you to mask out or reweight individual output timesteps, which is useful in sequence to sequence learning.

However, I happened to find that the sample_weight should have shape **(nb_samples, timesteps, 1)**, otherwise there will be some shape mismatch errors. 

If we do prefer **(nb_samples, timesteps)**, then there is a bug in:

https://github.com/fchollet/keras/blob/master/keras/models.py#L83-85

Because we need to expand the shape of sample_weight

**Comment by fchollet** [comment]
The shape of `sample_weight` is in fact expanded to match the expected size mentioned in the documentation: 
https://github.com/fchollet/keras/blob/master/keras/models.py#L98
https://github.com/fchollet/keras/blob/master/keras/models.py#L100

Can you clarify what you mean?

**Comment by blackyang** [comment]
Hi @fchollet , so sorry I just saw your reply...

I mean if I do provide a `sample_weight` in fit(), then it should be call first:
https://github.com/fchollet/keras/blob/master/keras/models.py#L85
then:
https://github.com/fchollet/keras/blob/master/keras/models.py#L20-25

but in the later function it doesn't expand `sample_weight` if `len(y.shape) > 1`

So a quick fix would be changing keras's documentation for `sample_weight` in `model.fit()` to:
>For time-distributed data, there is one weight per sample per timestep, i.e. if your output data is shaped (nb_samples, timesteps, output_dim), your mask should be of shape **(nb_samples, timesteps, 1)**. This allows you to mask out or reweight individual output timesteps, which is useful in sequence to sequence learning.

If using `sample_weight` of shape `(nb_samples, timesteps)` there would be an error

**Comment by fchollet** [comment]
It seems you're right. Would you like to submit a PR to update the documentation?

**Comment by blackyang** [comment]
Sure :)

Sorry for the late response again

**Comment by blackyang** [comment]
Hi @fchollet , the documentation on keras.io seems different from that on github, there is no `sample_weight` parameter in:
https://github.com/fchollet/keras/blob/master/docs/sources/models.md

I am afraid of potential conflict if I make a PR to update doc

**Comment by fchollet** [comment]
You can just update the Github version, the keras.io version will be synced later.

**Comment by blackyang** [comment]
Hi @fchollet , I closed the previous PR (my fork is far behind keras/master) and make a new PR
